dali/kernels/signal/fft/stft_gpu_impl.cu

```diff
@@ -0,0 +1,299 @@
+  }
+}
+
+void StftImplGPU::ReserveTempStorage(ScratchpadEstimator &se) {
+  // TODO(michalz) - try in-place transform to reduce memory footprint
+  // extracted windows
+  se.add<float>(AllocType::GPU, num_temp_windows() * transform_in_size(), 8);
+  // transform output
+  se.add<float2>(AllocType::GPU, num_temp_windows() * transform_out_size(), 8);
+
+  int windows_left = total_windows_;
+  int max_plan = num_temp_windows();
+
+  size_t total_work = 0;
+  while (windows_left > 0) {
+    auto it = plans_.upper_bound(max_plan);
+    assert(it != plans_.begin());
+    --it;
+    int batch = it->first;
+    total_work += it->second.work_size;
+    windows_left -= batch;
+    max_plan -= batch;
+  }
+  total_work_size_ = total_work;
+
+  se.add<char>(AllocType::GPU, total_work_size_, 16);
+}
+
+void StftImplGPU::ValidateParams(ExecutionContext &ctx) {
+  int N = ctx.in().num_samples();
+  const TensorListShape<2> &out_shape = ctx.output_shape();
+
+  DALI_ENFORCE(out_shape.num_samples() == N, make_string(
+        "Unexpected number of samples in output list: ", out_shape.num_samples(), " vs ", N));
+
+  for (int i = 0; i < N; i++) {
+    TensorShape<2> ts = transform_out_.shape[i];
+
+    if (!args_.time_major_layout)
+      std::swap(ts[0], ts[1]);
+
+    DALI_ENFORCE(out_shape[i] == ts,
+        make_string("Unexpected output shape at sample ", i, ": ", out_shape[i], " expected ", ts));
+  }
+
+  DALI_ENFORCE(ctx.window().num_elements() == 0 ||
+               ctx.window().num_elements() == transform_size(),
+               "The window must be either empty or have a size equal to the transform size.");
+
+}
+
+void StftImplGPU::RunR2C(KernelContext &ctx,
+                         const OutListGPU<complexf, 2> &out,
+                         const InListGPU<float, 1> &in,
+                         const InTensorGPU<float, 1> &window) {
+  assert(args_.spectrum_type == FFT_SPECTRUM_COMPLEX);
+
+  ExecutionContext ectx({ &ctx, &out, nullptr, &in, &window });
+  Run(ectx);
+}
+
+void StftImplGPU::Run(ExecutionContext &ctx) {
+  ValidateParams(ctx);
+  ExtractWindows(ctx);
+  RunTransform(ctx);
+  StoreResult(ctx);
+}
+
+void StftImplGPU::StoreResult(ExecutionContext &ctx) {
+  if (ctx.has_real_output())
+    StoreRealResult(ctx);
+  else
+    StoreComplexResult(ctx);
+}
+
+void StftImplGPU::StoreComplexResult(ExecutionContext &ctx) {
+  static_assert(sizeof(ctx.complex_out().data[0][0]) == sizeof(float2));
+  // reinterpret the whole TensorListView reference to avoid copying the shape
+  auto &out = reinterpret_cast<const OutListGPU<float2, 2> &>(ctx.complex_out());
+  post_complex_->Run(ctx.context(), out, transform_out_);
+}
+
+void StftImplGPU::StoreRealResult(ExecutionContext &ctx) {
+  post_real_->Run(ctx.context(), ctx.real_out(), transform_out_);
+}
+
+void StftImplGPU::RunTransform(ExecutionContext &ctx) {
+  float2 *fft_out = ctx.scratchpad()->Allocate<float2>(
+      AllocType::GPU, num_temp_windows() * transform_out_size(), 8);
```

Comment:
Why 8? I think the default value is alignof(T) -> alignof(float2), so default should do.